lang_GENERIC/analyze/Dataflow_constness.ml

```diff
@@ -0,0 +1,247 @@
+(*
+ * Copyright (C) 2020 r2c
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file license.txt.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * license.txt for more details.
+*)
+open Common
+open IL
+
+module G = AST_generic
+module F = IL
+module D = Dataflow
+module VarMap = Dataflow.VarMap
+
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+type kind =
```

Comment:
maybe type type_ = Tbool | Tint | ... ?
Kind usually means something else (the type of types).

Reply:
That's right, "kind" and "sort" are typically used to refer to the types of types themselves. I just wanted to avoid potential confusion with `AST_generic.type_`, but I can use another term too.